components/brave_wallet/browser/internal/hd_key_ed25519.cc

```diff
@@ -121,6 +121,11 @@ std::vector<uint8_t> HDKeyEd25519::GetPrivateKeyBytes() const {
   return {secret_key.begin(), secret_key.end()};
 }
 
+std::vector<uint8_t> HDKeyEd25519::GetPublicKeyBytes() const {
+  auto public_key = private_key_->unwrap().public_key_raw();
```

Comment:
I actually copypasted that from HDKeyEd25519::GetBase58EncodedPublicKey() assuming it was correct

Reply:
@kdenhartog that is intentional for us to catch unhandled errors and it is not calling rust `unwrap` but with `expect`
https://github.com/brave/brave-core/commit/9ae530a91777a74ae22e631ca73a913ad708f6dc#diff-137afd46973b9774a2b139da7f989776c8aeafa5684cd1e6951014366b695c20R28-R29 which was reviewed in
https://github.com/brave/security/issues/718#issuecomment-1018064995
And it is very unlikely to fail because the key is generated from [`generate_ed25519_extended_secrect_key_from_seed`](https://github.com/brave/brave-core/blob/2531782df4a916b84913b552084ff987660aeeda/components/brave_wallet/browser/internal/hd_key_ed25519.cc#L30-L32) and [`generate_ed25519_extended_secrect_key_from_bytes`](https://github.com/brave/brave-core/blob/2531782df4a916b84913b552084ff987660aeeda/components/brave_wallet/browser/internal/hd_key_ed25519.cc#L43-L45), both check `is_ok()`. Also guarded [`CHECK`](https://github.com/brave/brave-core/blob/2531782df4a916b84913b552084ff987660aeeda/components/brave_wallet/browser/internal/hd_key_ed25519.cc#L23) in constructor which is impossible to hit now but just to prevent someone from making bad changes in the future.

Reply:
Awesome, thanks for all the links to easily follow this. Definitely not concerned with this anymore